source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -36,33 +32,39 @@ ShaderGraph::ShaderGraph(const ShaderGraph* parent, const string& name, ConstDoc
 
 void ShaderGraph::addInputSockets(const InterfaceElement& elem, GenContext& context)
 {
-    for (ValueElementPtr port : elem.getActiveValueElements())
+    for (InputPtr nodeInput : elem.getActiveInputs())
     {
-        if (!port->isA<Output>())
+        ShaderGraphInputSocket* inputSocket = nullptr;
+        ValuePtr portValue = nodeInput->getResolvedValue();
+        if (!portValue)
         {
-            ShaderGraphInputSocket* inputSocket = nullptr;
-            const string& portValue = port->getResolvedValueString();
-            std::pair<const TypeDesc*, ValuePtr> enumResult;
-            const string& enumNames = port->getAttribute(ValueElement::ENUM_ATTRIBUTE);
-            const TypeDesc* portType = TypeDesc::get(port->getType());
-            if (context.getShaderGenerator().getSyntax().remapEnumeration(portValue, portType, enumNames, enumResult))
+            InputPtr interfaceInput = nodeInput->getInterfaceInput();
```

Comment:
Though we need this at the node level, I don't believe that it's possible for a nodegraph to have an Input with an "interfacename". I think I earlier has this logic as well. Probably best to not add this in unless there is a case for this?

Reply:
Good call, and I think you're correct that this case cannot occur.  I'll remove the test.